### DISCUSSION TITLE
Shadows Bug

### DISCUSSION THREAD

**Discussion by ghost** [CustomiesDevs/Customies#48] [Q&A] [answered]
Hello, sorry for the inconvenience. I succeded to create my own block xD!!! But now I have a problem, the shadows of my block is verry buggy
My basic texture: 
![image](https://user-images.githubusercontent.com/89030950/194303220-e8936533-a658-4a6a-8c24-e5a893de2bc3.png)

and the result:
![Capture d’écran (33)](https://user-images.githubusercontent.com/89030950/194302744-15ac799e-c45b-4669-b6b4-c6c464ef7453.png)

Thx for your previous helping.

**Comment by ghost** [comment]
@TwistedAsylumMC

**Comment by unickorn** [comment, marked as answer]
This looks like an issue with your texture. Can you try and see if there are any invisible blocks etc. that could cause this type of issue?

**Comment by ghost** [comment]
Thanks, it's working

**Comment by ghost** [comment]
But can you update the wiki with the new version of the plugin plz @Unickorn

**Comment by ghost** [comment]
Thank you Unickorn. Can I close the Issue ?

**Comment by unickorn** [comment]
As long as you no longer have an issue, you can elaborate how you fixed it and close it!

**Comment by ghost** [comment]
Ok, but now I have a new problem when I register my custom block (`ArgumentCountError: "Too few arguments to function customiesdevs\customies\util\Cache::__construct())`

**Comment by ghost** [comment]
I register the block like that


```
$material = new Material(Material::TARGET_ALL, "my_block", Material::RENDER_METHOD_ALPHA_TEST);

$model = new Model([$material], "geometry.my_block", new Vector3(-8,0, -8), new Vector3(16, 16, 16));

$creativeInfo = new CreativeInventoryInfo(CreativeInventoryInfo::CATEGORY_NATURE, CreativeInventoryInfo::GROUP_MISC_FOOD);

CustomiesBlockFactory::getInstance()->registerBlock(fn(int $id) => new Block(new BlockIdentifier(1501,0), "My Block", new BlockBreakInfo(1)), "server:my_block", $model,$creativeInfo);  
```

**Comment by ghost** [comment]
And the block class:

```


class MyBlock extends Opaque
{
    public function __construct()
    {
        parent::__construct(new BlockIdentifier(1501,0), "My Block", new BlockBreakInfo(1));

    }
}
```

**Comment by ghost** [comment]
@Unickorn

**Comment by unickorn** [comment]
Can you try updating your Customies to the latest version on poggit?

**Reply by ghost** [reply]
`Error: "Class "customiesdevs\customies\util\Cache" not found" (EXCEPTION) in "plugins/Customies-master/src/Customies" at line 17`

**Comment by unickorn** [comment]
Try this dev build for now, that was a bug fixed after the release. https://poggit.pmmp.io/r/194948/Customies_dev-29.phar

**Reply by ghost** [reply]
ok

**Comment by ghost** [comment]
it's the same bug @Unickorn

**Comment by ghost** [comment]
`[AsyncWorker#0 thread/CRITICAL]: ArgumentCountError: "Too few arguments to function customiesdevs\customies\util\Cache::__construct()`

**Reply by ghost** [reply]
@Unickorn

**Reply by unickorn** [reply]
This does seem to be a bug, let's leave this discussion at that and deal with this on another issue. Please create a new one!

**Reply by ghost** [reply]
ok

**Reply by ghost** [reply]
https://github.com/CustomiesDevs/Customies/issues/50
